### ISSUE TITLE
URL links to dashboard are not loading the widgets

### ISSUE THREAD

**Issue by idan-marciano-jfrog** [reportportal/reportportal#1790]
When sharing a dashboard that contains multiple multiple widget with a link to it, For example
`https://reportportal.company.org/ui/#project/dashboard/6059`
The widgets are not been loaded and an empty dashboard is been displayed.
When entering the same dashboard from the dashboard list its been loaded as expected.
From looking at the network requests when i enter the link it seems to be lacking the widgets requests and only the dashboard id request is been loaded to the page, When looking at the same request while browsing from dashboard list it seems like there are separate requests for all the widgets that are contained in the specific dashboard.  
Refreshing the page is not helping, only browsing from dashboard list display all data

**Versions:**
 - MAC OS Monterey 12.4
 - CHROME Version 103.0.5060.134 (Official Build) (x86_64)
 - Version of RP :
 Current version:API Service: 5.6.2;Jobs Service: 5.6.2;Authorization Service: 5.6.2;Service UI: 5.6.0;

**Comment by DzmitryHumianiuk** [comment]
@idan-marciano-jfrog please check if the dashboard is shared across the project (right column on the pic).
If not - other users won't have access to it.

<img width="1224" alt="image" src="https://user-images.githubusercontent.com/11332788/181658067-a638ad26-ba30-417b-90ca-c73990c23924.png">

**Comment by idan-marciano-jfrog** [comment]
@DzmitryHumianiuk Yes it it shared.
Were you not able to reproduce?

**Comment by VolhaKarenko** [comment]
Hi @idan-marciano-jfrog, could you please try refreshing the page after following the link, are the widgets showing up? 
I did reproduce the issue but refreshing the page works - this is a known issue for us, not always reproducible, we already have it in the backlog.

**Comment by idan-marciano-jfrog** [comment]
@VolhaKarenko no, for me refreshing didnt help i see an empty dashboard without any widgets 
![Screen Shot 2022-09-04 at 11 22 10](https://user-images.githubusercontent.com/68947907/188304434-23ba490d-6fda-4b82-a083-8f4bf81ebefe.png)

**Comment by VolhaKarenko** [comment]
I could not reproduce the issue on version 5.7.2. @idan-marciano-jfrog, are there any errors in the console when following the dashboard link? Please share the response for the dashboard id request.

**Comment by idan-marciano-jfrog** [comment]
@VolhaKarenko From looking at the network request on the dashboard it seems like there is no error 
The content of the request is

`{"description":"This dashboard shows Artifactory automation results for version: draft-full-7.45.0-m020","owner":"artifactory_auto","share":true,"id":7474,"name":"draft-full-7.45.0-m020","widgets":[{"widgetName":"project_component_health_draft-full-7.45.0-m020_1662337106439","widgetId":7465,"widgetType":"componentHealthCheck","widgetSize":{"width":12,"height":6},"widgetPosition":{"positionX":0,"positionY":6},"share":true},{"widgetName":"overall-version-health_draft-full-7.45.0-m020_1662337196646","widgetId":7469,"widgetType":"overallStatistics","widgetSize":{"width":6,"height":6},"widgetPosition":{"positionX":0,"positionY":0},"share":true},{"widgetName":"pit_pm_health_draft-full-7.45.0-m020_1662337171003","widgetId":7467,"widgetType":"componentHealthCheck","widgetSize":{"width":6,"height":5},"widgetPosition":{"positionX":0,"positionY":12},"share":true},{"widgetName":"sanity-health_draft-full-7.45.0-m020_1662337272124","widgetId":7473,"widgetType":"componentHealthCheck","widgetSize":{"width":6,"height":5},"widgetPosition":{"positionX":6,"positionY":12},"share":true},{"widgetName":"top-20-project-patterns_draft-full-7.45.0-m020_1662337221997","widgetId":7471,"widgetType":"topPatternTemplates","widgetSize":{"width":6,"height":6},"widgetPosition":{"positionX":6,"positionY":0},"share":true}]}`
I attached a video
https://user-images.githubusercontent.com/68947907/188509269-f0d3dbc0-1dee-4693-adbe-2bbbf3425996.mov

**Comment by VolhaKarenko** [comment]
@idan-marciano-jfrog Thank you! We have created a separate task to investigate the issue, as the root cause is not clear now. We will keep you updated on it.

**Comment by VolhaKarenko** [comment]
Hello @idan-marciano-jfrog, we have tried various scenarios, but unfortunately, we were unable to reproduce this issue. Since there is a workaround (opening the dashboard from the dashboard list), I would close the issue. The team is currently busy with higher priority tasks, and I can't promise you that we'll get back to this problem anytime soon.
I could advise you to try to upgrade to the latest version 5.7.2 and check this behavior once again

**Comment by idan-marciano-jfrog** [comment]
@VolhaKarenko I managed to understand when this scenario happen. It occurs when there are few hundreds of dashboards.
When there are the ui is not able to load the dashboard widgets, When i clean the dashboard it seems like the problems is resolved but will reproduce when ill reach this "limit" again.

**Comment by VolhaKarenko** [comment]
@idan-marciano-jfrog, thanks for the useful details! 
We have a limit of 300 dashboards per project. I managed to reproduce the issue when I create more than 300 dashboards, it seems like the dashboard is created, but actually, I can not see it on the Dashboards tab - we should certainly add a validation from the UI side. It is also not possible to add widgets on a newly created dashboard if there are more than 300 dashboards in total.
Could you clarify if all the created dashboards are available in your case and if widgets can be added successfully?

**Comment by idan-marciano-jfrog** [comment]
@VolhaKarenko I am happy to hear that i helped.
few questions:
1. @DzmitryHumianiuk Is this limitation documented? i could not find it.
2. @DzmitryHumianiuk Is paying costumers also have this limitation?
3. Why the filters, dashboard and widgets has the same counter ID? its confusing.
In my case i will need to add some kind of cleanup cron job that will clean the old dashboards using API's. Like i mentioned when i create  a dashboard i also create few widgets and filters meaning that if i create dashborad with id 100 and 1 widget and 1 filter the next dashboard will be ID 103. This is kind of bad implementation logic because those are different objects and when i want to delete dashboard 100-200 via special scripts it will only delete 33 dashboards beacuse the other id's are not dashboards and the API will fail.
4. When i delete dashboard its also deleting all its content meaning the widget that are assigned to it and same goes to widgets and assigned widgets filter. Back to the above example when i delete dashboard 100 will it delete 101 and 102 objects as well?

**Comment by DzmitryHumianiuk** [comment]
@idan-marciano-jfrog 
1. it's not supposed to be a limitation at all. I mean, at the implementation stage we didn't consider that anyone will create this amount of dashboards and show them on UI.
2. it is the same for anyone. Just need to update UI app to address this.

> one of the possible workarounds might be deletion of expired dashboards.

3. Good catch, will check it with the team. (it might be same object from API perspective)

4. Yes, it should delete all nested objects, except of filters. Thus, dashboard and widgets.


**In general, 
this pushes me to think, was it worth it to recreate such a number of new dashboards and widgets at all?**

I mean, filters are intended to be replaceable source of data. And should be used for this reason.
If you have a dashboard and widget which represents required data based on particular selection made with filter, then it's much easier to rewrite just the filter.


```bash
curl 'https://{SERVER}/api/v1/{PROJECT}/filter/{FILTER_ID}' \
  -X 'PUT' \
  --data-raw '{
  "owner": "{OWNER}",
  "share": false,
  "name": "{FILTER_NAME}",
  "conditions": [
    {
      "filteringField": "compositeAttribute",
      "condition": "has",
      "value": "XX:YYY",
      "attributes": [
        {
          "key": "XX",
          "value": "YYY"
        }
      ]
    }
  ],
  "orders": [
    {
      "sortingColumn": "startTime",
      "isAsc": false
    },
    {
      "sortingColumn": "number",
      "isAsc": false
    }
  ],
  "type": "Launch"
}' 
```

Take a look what is the UI sending when you save a filter.

Just play with the filters on UI by combining various attributes and conditions together.
and use it for the API calls.

<img width="1060" alt="image" src="https://user-images.githubusercontent.com/11332788/203680147-0e6aa75d-79ba-4078-9f08-92dec01111fd.png">

**Comment by VolhaKarenko** [comment]
@idan-marciano-jfrog regarding the limitation, unfortunately, it is not documented anywhere, but we actually have increased the number of dashboards by 300 in version 5.4 after this issue [1199](https://github.com/reportportal/reportportal/issues/1199) 
This is the max number we can allow right now from the performance point of view. The problem I see now is that we are not properly validating the request to create 301+ dashboards, it should not be allowed both on UI or via API to avoid cases as you mentioned. 
Thank you once again for sharing this with us!

**Comment by DzmitryHumianiuk** [comment]
@idan-marciano-jfrog did you have a chance to try my solution?

**Comment by alena-babrova** [comment]
Dear @idan-marciano-jfrog,
Could you please tell me if your question is still relevant?

**Comment by idan-marciano-jfrog** [comment]
@alena-babrova From my side i still experience the 300 dashboard limitation. not the best experience. 
My internal solution is to run cleanup job every once in a while.

**Comment by DzmitryHumianiuk** [comment]
@idan-marciano-jfrog did you try this: https://github.com/reportportal/reportportal/issues/1790#issuecomment-1325867704 ?

**Comment by DzmitryHumianiuk** [comment]
the same: https://github.com/reportportal/reportportal/issues/2168
